prov/util/src/util_coll.c

```diff
@@ -199,23 +208,30 @@ static inline int util_coll_mc_alloc(struct util_coll_mc **coll_mc)
 
 	slist_init(&(*coll_mc)->barrier_list);
 	slist_init(&(*coll_mc)->deferred_list);
+	slist_init(&(*coll_mc)->pending_xfer_list);
 	return FI_SUCCESS;
 }
 
-/* TODO: see ofi.h for power of 2 helper */
-static inline int util_coll_pof2(int num)
+static inline uint64_t util_coll_form_tag(uint32_t coll_id, uint32_t rank)
 {
-	int pof2 = 1;
+	uint64_t tag;
+	uint64_t src_rank = rank;
+
+	tag = coll_id;
+	tag |= (src_rank << OFI_COLL_TAG_SOURCE_SHIFT);
```

Comment:
This define is only used here, and since it's tied to the u32 parameters, maybe we should just hard code 32 here.